src/libsyntax/parse/parser.rs

```diff
@@ -2663,7 +2665,7 @@ impl<'a> Parser<'a> {
     }
 
     pub fn check_unknown_macro_variable(&mut self) {
-        if self.quote_depth == 0 {
+        if self.quote_depth == 0 && !self.parsing_token_tree {
```

Comment:
I don't think so -- looking into it some more, the problem is that [this condition](https://github.com/rust-lang/rust/blob/master/src/libsyntax/parse/parser.rs#L2813) needs to be false when parsing the lhs of a `macro_rules!` body.